### ISSUE TITLE
Revisit `sample_size` and `batch_size`

### ISSUE THREAD

**Issue by fehiepsi** [ibab/tensorflow-wavenet#179]
According to ByteNet [paper](https://arxiv.org/pdf/1610.10099v1.pdf), `batch_size > 1`, `sample_size ~ 2 x receptive field`, and the loss is calculated ignoring the first receptive field.

> ByteNet Decoder that we use for the result has 25 residual blocks split into five sets of five blocks each; for the five blocks in each set the dilation rates are, respectively, 1,2,4,8 and 16. The masked kernel has size 3. This gives a receptive field of 315 characters. The number of hidden units d is 892. For this task we use residual multiplicative blocks and Sub-BN (Fig. 3 Right); we do not use bags of character n-grams for the inputs. For the optimization we use Adam (Kingma and Ba, 2014) with a learning rate of 10−2 and a weight decay term of 10−5. We do not reduce the learning rate during training. At each step we sample a batch of sequences of 515 characters each, use the first 315 characters as context and predict only the latter 200 characters.

With current wavenet network, I think that the reason we use `batch_size=1` is to reduce memory. And the reason we calculate the loss including the first receptive field is because the `sample_size` is large, then the effect of the first receptive field is small. Am I right? Could anyone please help me understand these parameters better?

**Comment by fehiepsi** [comment]
We have discussion here and there about this problem already. No need to create a new one, so I close.
